ui/job-view/details/tabs/PerformanceTab.jsx

```diff
@@ -16,6 +86,63 @@ export default class PerformanceTab extends React.PureComponent {
         role="region"
         aria-label="Performance"
       >
+        <div className="performance-panel-actions">
+          {
+            // If there is a profiler link, show this first. This is most likely
+            // the primary action of the user here.
+            profilerLink
+          }
+          {
+            // Just to be safe, use the same isPerfTest check the other
+            // "Create Gecko Profile" button uses in the action menu.
```

Comment:
There's quite a few comments here but I'm not sure they're really necessary :)

Reply:
Explaining the context for why code exists is important, especially maintaining code over the long term. I like to encode the decision and reasoning behind different actions in the code. 

```
/**
 * The performance tab shows performance-oriented information about a test run.
 * It helps users interact with the Firefox Profiler, and summarizes test
 * timing information.
 */
```

The component pre-amble explains why a component exists, what function it's serving, and why it's important.

```
// If there is a profiler link, show this first. This is most likely
// the primary action of the user here.
```

I used the primary blue here for the color, and want to let future readers know my intent for why this button is important. From a programming perspective, it doesn't what order components the buttons come in, but I would like to preserve the importance of keeping the button as a primary action. From a workflow perspective, putting something else first could regress it.

```
// Just to be safe, use the same isPerfTest check the other
// "Create Gecko Profile" button uses in the action menu.
```

For this check, I don't think it's completely necessary, but I want to be safe in case I present an action that the user can't perform.

Reply:
> Explaining the context for why code exists is important, especially maintaining code over the long term. I like to encode the decision and reasoning behind different actions in the code.

I do understand the importance of using comments to explain context where it cannot be explained easily via variable or function naming. And I recognize that everyone has a different preference as to their level of documentation via comments. :) To be more specific in my original comment, comments like this strike me as redundant and make the code more cluttered.

```<Button
     className={`btn ${
         // Only make this primary if there is no profiler link.
                  profilerLink
                    ? 'btn-outline-darker-secondary'
                    : 'btn-darker-secondary' (was  `primary` in previous commit)
                } btn-sm`}
```

So it was a nit, to be sure, but also a poorly worded request to please omit comments like this and only include comments that are truly necessary for context. :)

Also, apologies if my last comment was confusing regarding the primary blue. I was not trying to imply you couldn't use any color at all. We tend to use `btn-darker-info` cyan/turquoise rather than the primary blue color, so for consistency, that is why I mentioned it. The info color is definitely Ok to use to announce the importance of the button.

Reply:
> I don't think this work blocks the upcoming push health work either, and as these are React components, it should be pretty easy to shift things around to handle the new way of doing things once users are on-boarded to the new flow. I'd be happy to follow-up with that work. My concern on not handling these workflow issues now, is that there is a fairly urgent need to have great performance tooling, where it's easy for any Firefox engineer to debug their performance issues.

I don't think we need to wait until push health is 100% complete (which will be an evolving process anyways) to start adding some of your features, so my comment was not meant to imply that this pr should wait. Ideally new features, where it makes sense, will be added to both. And at some point we can figure out when it'd make sense to only add them to Push Health. There are some shared components between Treeherder and Push Health but because of differences between sheriff and developer needs, not everything that looks like it would be shared actually is. Could you also add these changes to the push health file I mentioned in my comment? 

We will be following up with our original group of users soon - you included ;) - to get feedback on the push health changes and I think having these changes included would be really useful.